### PR TITLE
[CosmosDB] Add change for long running operations

### PR DESCRIPTION
**Pull request by kristynhamasaki**
Small change that fixes long running operations returning unauthorized.

### CODE REVIEW

File: sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/src/Generated/SqlResourcesOperations.cs [hunk 1]
```diff
@@ -3569,7 +3570,11 @@ public async Task<AzureOperationResponse<SqlRoleDefinitionGetResults>> GetSqlRol
             HttpStatusCode _statusCode = _httpResponse.StatusCode;
             cancellationToken.ThrowIfCancellationRequested();
             string _responseContent = null;
-            if ((int)_statusCode != 200)
+            if (!((int)_statusCode != 200 && (int)_statusCode != 202))
+            {
+                _httpRequest.RequestUri = _originRequestUri;
+            }
+            if ((int)_statusCode != 200 && (int)_statusCode != 202)
```
Comment (MehaKaushik) [inline sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/src/Generated/SqlResourcesOperations.cs:3577]: if ((int)_statusCode != 200)

File: sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/src/Generated/SqlResourcesOperations.cs [hunk 2]
```diff
@@ -3569,7 +3570,11 @@ public async Task<AzureOperationResponse<SqlRoleDefinitionGetResults>> GetSqlRol
             HttpStatusCode _statusCode = _httpResponse.StatusCode;
             cancellationToken.ThrowIfCancellationRequested();
             string _responseContent = null;
-            if ((int)_statusCode != 200)
+            if (!((int)_statusCode != 200 && (int)_statusCode != 202))
```
Comment (MehaKaushik) [inline sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/src/Generated/SqlResourcesOperations.cs:3573]: if (!((int)_statusCode != 200)

**Review by MehaKaushik — APPROVED** [review]
lgtm

File: sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/src/Generated/SqlResourcesOperations.cs (hunk 1 above)
Comment (kristynhamasaki) [reply sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/src/Generated/SqlResourcesOperations.cs]: done + fixed other mismatched codes. thanks!

File: sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/src/Generated/SqlResourcesOperations.cs (hunk 2 above)
Comment (kristynhamasaki) [reply sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/src/Generated/SqlResourcesOperations.cs]: done

**Comment by kristynhamasaki** [comment]
Hi @allenjzhang please take a look at this PR to add change for long running operations, which fixes the Unauthorized result we are currently getting. Thank you!

**Review by markcowl — CHANGES_REQUESTED** [review]
Was this PR the result of running the generation script, or are these by-hand edits to generated code?

If there was an associated Swagger change, please include a link the swagger PR in the description, and include the management metadata file created by the generator in the /eng folder.

  If not, we will have to consider how to make this change, as changing generated code is not allowed (among other things, the changes will be erased the next time the api is updated)

**Comment by kristynhamasaki** [comment]
Hi @markcowl yes this is not part of the generated code but a fix for this [issue](https://github.com/Azure/azure-sdk-for-net/issues/10639). Meha has made a permanent change for this [here](https://msdata.visualstudio.com/DefaultCollection/CosmosDB/_git/CosmosDB/pullrequest/438858) but it is not deployed to Prod yet, so we would like to make this change in the mean time so we can release our PS cmdlets to customers. Thanks!

**Review by markcowl — APPROVED** [review]

**Comment by markcowl** [comment]
Talked offline.  There is a service-side fix currently being deployed, this is a stopgap until that deployment completes.  The service team has agreed not to update the api until the fix is deployed.

**Pull request merged**